Jenkinsfile

```diff
@@ -1,5 +1,9 @@
 properties([disableConcurrentBuilds(abortPrevious: true), buildDiscarder(logRotator(numToKeepStr: '7'))])
 
+if (BRANCH_NAME == 'master' && currentBuild.buildCauses*._class == ['jenkins.branch.BranchEventCause']) {
+  error 'No longer running builds on response to master branch pushes. If you wish to cut a release, use “Re-run checks” from this failing check in https://github.com/jenkinsci/bom/commits/master'
```

Comment:
why do we need to error? can't we just skip the build with a return?

the build already passed on the PR before merging.

Reply:
do we need a failing check? you can create a pull request to test a build and it won't be on master if it hasn't passed on a pull request.

it's not ideal to open the repository and just see red.

![image](https://user-images.githubusercontent.com/21194782/234330416-9ae2f022-0354-40a5-a5d7-19e9b8438513.png)

Reply:
what I mean though is why do we need to run a successful build of bom on master? i.e. with branch protection configured here it assures that it passed on the PR build and there's nothing different in this repo between PR and master can't we just auto release on master if labels match?

Reply:
Oh I see. That would be possible. There are two considerations though:
* This PR ensures that the remaining `master` builds test all LTS branches, so it is actually safer than the prior setup. To replicate that safety with your proposed setup, the onus would be on whoever merged a PR with an “interesting” label to remember to run a build with the `full-test` label first.
* Since Jenkins PR-merge strategy is long since disabled, and we are not using a GH merge queue, a passing PR-head build does not guarantee that all is well after the merge. It would then become possible for a `bom` release to be cut from a revision which is actually broken. We would notice pretty soon (when new DB PRs start consistently failing with the same error with no apparent relation to the diff) and issue a revert or correction, but it still makes me nervous. (For the same reason, the JEP-229 workflow generally checks `master` status, even though we expect it “should” be passing.)

I have no strong opinion, so if you want to experiment with another policy, go ahead.